code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -935,6 +942,58 @@ func (sa *StorageAllocation) changeBlobbers(
 	return blobbers, nil
 }
 
+func removeBlobberAllocationFromChallPartition(
+	removeId string,
+	allocID string,
+	allocPartitionIndex int,
+	ssc *StorageSmartContract,
+	balances chainstate.StateContextI) error {
+	blobberAllocChallPartition, err := getBlobbersChallengeAllocationList(removeId, balances)
+	if err != nil {
+		return fmt.Errorf("cannot fetch blobber allocation partition: %v", err)
+	}
+	err = blobberAllocChallPartition.RemoveItem(balances, allocPartitionIndex, allocID)
+	if err != nil {
+		return fmt.Errorf("error removing allocation from challenge partition: %v", err)
+	}
+
+	blobberAllocChallSize, err := blobberAllocChallPartition.Size(balances)
+	if err != nil {
+		return fmt.Errorf("error getting size of challenge partition: %v", err)
+	}
+	if blobberAllocChallSize-1 <= 0 {
```

Comment:
The blobberAllocChallSize is the total item number in the partitions, so if all have been removed, it should be value of 0, not 1.

Reply:
If we're doing Save at the end. It will be 1.. 
as it gets data from MPT and does size.
Since our partition is not saved yet. The size will be same as old one..

Reply:
But in case if its 0 it should be 0. Yea you're right